tests/system/providers/amazon/aws/example_sagemaker.py

```diff
@@ -161,8 +161,12 @@ def _build_and_upload_docker_image(preprocess_script, repository_uri):
 
         docker_build_and_push_commands = f"""
             cp /root/.aws/credentials /tmp/credentials &&
+            # login to public ecr repo containing amazonlinux image
+            docker login --username {username} --password {password} public.ecr.aws
```

Comment:
Just a question do we actually need to login to public ECR? Or it is also has some limitation?

```console

❯ docker pull public.ecr.aws/amazonlinux/amazonlinux:latest
latest: Pulling from amazonlinux/amazonlinux
1b843b796f14: Pull complete 
Digest: sha256:6fef13bccd69e4e70a257ba03c6476e8c8007510478d574f69ce9f80751ee9be
Status: Downloaded newer image for public.ecr.aws/amazonlinux/amazonlinux:latest
public.ecr.aws/amazonlinux/amazonlinux:latest
```

Reply:
oh well maybe we don't need to ? I followed the instructions on https://docs.aws.amazon.com/AmazonECR/latest/userguide/amazon_linux_container_image.html which are to 1) login 2) pull.
Since we have the credentials laying around here, it's not much effort, so I'd keep it ?

Reply:
found some info here: https://aws.amazon.com/blogs/aws/amazon-ecr-public-a-new-public-container-registry/
```Anyone who pulls images anonymously will get 500 GB of free data bandwidth each month, after which they can sign up or sign in to an AWS account to get more. Simply authenticating with an AWS account increases free data bandwidth up to 5 TB each month when pulling images from the internet.```

AmazonLinux is a ~100MB download, so we can pull it 5000 times a month, or roughly 7 times/hour without being authenticated.
BUT the problem we have is that usually those unauthenticated quotas are applied per IP, and those IPs can be reused, so we're not the only ones on the quota.
I think it's just safer to login and not worry about it.

Reply:
Oh... I no idea why it required login. Maybe it is for compatibility with regular ECR. Or some kind of autogenerated documentation.

Anyway I don't think it affect anything, so we can keep it if docker apply creds from private ECR to public ECR.

Reply:
Oh... thanks for info!